Windows build is failing

From https://github.com/pystatgen/sgkit/actions/runs/5850622797/job/15860289990#step:5:4585

```
============================= test session starts =============================
platform win32 -- Python 3.8.17, pytest-7.4.0, pluggy-1.2.0 -- C:\Miniconda3\envs\test\python.exe
cachedir: .pytest_cache
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('D:\\a\\sgkit\\sgkit\\.hypothesis\\examples')
rootdir: D:\a\sgkit\sgkit
configfile: setup.cfg
plugins: hypothesis-6.82.4, cov-4.1.0, datadir-1.4.1, mock-3.11.1
collecting ... collected [10](https://github.com/pystatgen/sgkit/actions/runs/5633732790/job/15262949106#step:5:11) items / [12](https://github.com/pystatgen/sgkit/actions/runs/5633732790/job/15262949106#step:5:13)7 errors

=================================== ERRORS ====================================
_____________________ ERROR collecting sgkit/__init__.py ______________________
C:\Miniconda3\envs\test\lib\site-packages\_pytest\runner.py:341: in from_call
    result: Optional[TResult] = func()
C:\Miniconda3\envs\test\lib\site-packages\_pytest\runner.py:372: in <lambda>
    call = CallInfo.from_call(lambda: list(collector.collect()), "collect")
C:\Miniconda3\envs\test\lib\site-packages\_pytest\doctest.py:547: in collect
    module = import_path(
C:\Miniconda3\envs\test\lib\site-packages\_pytest\pathlib.py:565: in import_path
    importlib.import_module(module_name)
C:\Miniconda3\envs\test\lib\importlib\__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:10[14](https://github.com/pystatgen/sgkit/actions/runs/5633732790/job/15262949106#step:5:15): in _gcd_import
    ???
<frozen importlib._bootstrap>:991: in _find_and_load
    ???
<frozen importlib._bootstrap>:975: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:671: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:843: in exec_module
    ???
<frozen importlib._bootstrap>:2[19](https://github.com/pystatgen/sgkit/actions/runs/5633732790/job/15262949106#step:5:20): in _call_with_frames_removed
    ???
sgkit\__init__.py:[29](https://github.com/pystatgen/sgkit/actions/runs/5633732790/job/15262949106#step:5:30): in <module>
    from .stats.association import gwas_linear_regression, regenie_loco_regression
sgkit\stats\association.py:6: in <module>
    from dask.array import Array, stats
C:\Miniconda3\envs\test\lib\site-packages\dask\array\stats.py:[45](https://github.com/pystatgen/sgkit/actions/runs/5633732790/job/15262949106#step:5:46): in <module>
    from scipy.stats.stats import (
E   ImportError: cannot import name 'Ttest_1sampResult' from 'scipy.stats.stats' (C:\Miniconda3\envs\test\lib\site-packages\scipy\stats\stats.py)
```

It looks like it is installing an old version of Dask (dask-2022.2.1), which may be causing problems?

With a later version of Dask (2023.8.1) the error we get is the following. But this is only on Windows when running Python 3.9 which installs Pandas 2 (Python 3.8 uses Pandas 1).

```
___________________ test_zarr_to_plink[example_with_fam- ] ____________________

shared_datadir = WindowsPath('C:/Users/runneradmin/AppData/Local/Temp/pytest-of-runneradmin/pytest-0/test_zarr_to_plink_example_wit0/data')
tmp_path = WindowsPath('C:/Users/runneradmin/AppData/Local/Temp/pytest-of-runneradmin/pytest-0/test_zarr_to_plink_example_wit0')
plink_in = 'example_with_fam', fam_sep = ' '

    @pytest.mark.parametrize(
        "plink_in, fam_sep",
        [
            (example_dataset_1, "\t"),
            (example_dataset_2, " "),
            (example_dataset_3, " "),
        ],
    )
    def test_zarr_to_plink(shared_datadir, tmp_path, plink_in, fam_sep):
        # read plink file as a zarr file then write it out again
        zarr_path = tmp_path / "plink.zarr"
>       plink_to_zarr(path=shared_datadir / plink_in, output=zarr_path, fam_sep=fam_sep)

sgkit\tests\io\plink\test_plink_writer.py:66: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
sgkit\io\plink\plink_reader.py:378: in plink_to_zarr
    ds = read_plink(
sgkit\io\plink\plink_reader.py:256: in read_plink
    arr_fam = dataframe_to_dict(df_fam, dtype=FAM_ARRAY_DTYPE)
sgkit\io\utils.py:31: in dataframe_to_dict
    a = df[c].to_dask_array(lengths=True)
C:\Miniconda3\envs\test\lib\site-packages\dask\dataframe\core.py:1877: in to_dask_array
    arr = self.values
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = Dask Series Structure:
npartitions=1
    string
       ...
Name: family_id, dtype: string
Dask Name: getitem, 2 graph layers

    @property
    def values(self):
        """Return a dask.array of the values of this dataframe
    
        Warning: This creates a dask.array without precise shape information.
        Operations that depend on shape information, like slicing or reshaping,
        will not work.
        """
        if is_extension_array_dtype(self._meta.values):
>           warnings.warn(
                "Dask currently has limited support for converting pandas extension dtypes "
                f"to arrays. Converting {self._meta.values.dtype} to object dtype.",
                UserWarning,
            )
E           UserWarning: Dask currently has limited support for converting pandas extension dtypes to arrays. Converting string to object dtype.

--- a/.github/workflows/windows.yml
+++ b/.github/workflows/windows.yml
@@ -14,7 +14,7 @@ jobs:
     runs-on: windows-latest
     strategy:
       matrix:
-        python-version: ["3.8", "3.9"]
+        python-version: ["3.9"]
 
     steps:
     - uses: actions/checkout@v2

--- a/docs/examples/gwas_tutorial.ipynb
+++ b/docs/examples/gwas_tutorial.ipynb
@@ -5878,7 +5878,8 @@
     "sample_dp_mean = dp.mean(dim=\"variants\")\n",
     "sample_dp_mean.attrs[\"long_name\"] = \"Mean Sample DP\"\n",
     "ds[\"sample_dp_mean\"] = sample_dp_mean # add new data array to dataset\n",
-    "ds.plot.scatter(x=\"sample_dp_mean\", y=\"sample_call_rate\", size=8, s=10);"
+    "# Following does not work with recent versions of xarray, see https://github.com/pystatgen/sgkit/issues/934\n",
+    "#ds.plot.scatter(x=\"sample_dp_mean\", y=\"sample_call_rate\", size=8, s=10);"
    ]
   },
   {
@@ -13116,7 +13117,8 @@
     "ds_pca[\"sample_pca_projection_0\"] = ds_pca.sample_pca_projection[:,0]\n",
     "ds_pca[\"sample_pca_projection_1\"] = ds_pca.sample_pca_projection[:,1]\n",
     "ds_pca[\"sample_pca_projection_2\"] = ds_pca.sample_pca_projection[:,2]\n",
-    "ds_pca.plot.scatter(x=\"sample_pca_projection_0\", y=\"sample_pca_projection_1\", hue=\"SuperPopulation\", size=8, s=10);"
+    "# Following does not work with recent versions of xarray, see https://github.com/pystatgen/sgkit/issues/934\n",
+    "#ds_pca.plot.scatter(x=\"sample_pca_projection_0\", y=\"sample_pca_projection_1\", hue=\"SuperPopulation\", size=8, s=10);"
    ]
   },
   {
@@ -13223,7 +13225,7 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.8.16"
+   "version": "3.10.12"
   }
  },
  "nbformat": 4,

--- a/docs/examples/gwas_tutorial.ipynb
+++ b/docs/examples/gwas_tutorial.ipynb
@@ -5904,7 +5904,7 @@
     }
    ],
    "source": [
-    "ds = ds.sel(samples=((ds.sample_dp_mean >= 4) & (ds.sample_call_rate >= 0.97)))\n",
+    "ds = ds.sel(samples=((ds.sample_dp_mean >= 4) & (ds.sample_call_rate >= 0.97)).compute())\n",
     "print(f\"After filter, {len(ds.samples)}/284 samples remain.\")"
    ]
   },
@@ -9219,7 +9219,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "ds = ds.sel(variants=((ds.variant_allele_frequency[:,1] > 0.01) & (ds.variant_hwe_p_value > 1e-6)))"
+    "ds = ds.sel(variants=((ds.variant_allele_frequency[:,1] > 0.01) & (ds.variant_hwe_p_value > 1e-6)).compute())"
    ]
   },
   {
@@ -11181,8 +11181,8 @@
    "source": [
     "# To run PCA we need to filter out variants with any missing alt allele counts\n",
     "# Or where the counts are zero for all samples\n",
-    "variant_mask = ((ds_pca.call_alternate_allele_count < 0).any(dim=\"samples\")) | \\\n",
-    "    (ds_pca.call_alternate_allele_count.std(dim=\"samples\") <= 0.0)\n",
+    "variant_mask = (((ds_pca.call_alternate_allele_count < 0).any(dim=\"samples\")) | \\\n",
+    "    (ds_pca.call_alternate_allele_count.std(dim=\"samples\") <= 0.0)).compute()\n",
     "ds_pca = ds_pca.sel(variants=~variant_mask)"
    ]
   },

--- a/docs/how_do_i.rst
+++ b/docs/how_do_i.rst
@@ -135,11 +135,11 @@ Call :meth:`xarray.Dataset.sel` on the ``variants`` dimension:
 .. ipython:: python
 
     ds2 = sg.hardy_weinberg_test(ds)
-    ds2.sel(variants=(ds2.variant_hwe_p_value > 1e-2))
+    ds2.sel(variants=(ds2.variant_hwe_p_value > 1e-2).compute())
 
 .. note::
 
-   Filtering causes an eager Xarray computation.
+   The call to ``compute`` is needed to avoid an Xarray error.
 
 Find which new variables were added by a method?
 ------------------------------------------------

--- a/requirements-doc.txt
+++ b/requirements-doc.txt
@@ -8,5 +8,3 @@ matplotlib
 graphviz
 seaborn
 ablog!=0.10.27
-#Pinned due to #934
-xarray<=2022.9.0

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,7 +1,7 @@
 numpy
 xarray
-dask[array] >= 2022.01.0
-distributed >= 2022.01.0
+dask[array] >= 2023.01.0
+distributed >= 2023.01.0
 dask-ml
 scipy
 typing-extensions

--- a/sgkit/io/bgen/bgen_reader.py
+++ b/sgkit/io/bgen/bgen_reader.py
@@ -273,50 +273,53 @@ def read_bgen(
             f"`contig_dtype` must be of string or int type, not {contig_dtype}"
         )
 
-    path = Path(path)
-    sample_path = Path(sample_path) if sample_path else path.with_suffix(".sample")
+    with dask.config.set({"dataframe.convert-string": False}):
+        path = Path(path)
+        sample_path = Path(sample_path) if sample_path else path.with_suffix(".sample")
 
-    if sample_path.exists():
-        sample_id = read_samples(sample_path).sample_id.values.astype("U")
-    else:
-        sample_id = _default_sample_ids(path)
+        if sample_path.exists():
+            sample_id = read_samples(sample_path).sample_id.values.astype("U")
+        else:
+            sample_id = _default_sample_ids(path)
 
-    bgen_reader = BgenReader(path, metafile_path=metafile_path, dtype=gp_dtype)
+        bgen_reader = BgenReader(path, metafile_path=metafile_path, dtype=gp_dtype)
 
-    df = read_metafile(bgen_reader.metafile_path)
-    if persist:
-        df = df.persist()
-    arrs = dataframe_to_dict(df, METAFILE_DTYPE)
+        df = read_metafile(bgen_reader.metafile_path)
+        if persist:
+            df = df.persist()
+        arrs = dataframe_to_dict(df, METAFILE_DTYPE)
 
-    variant_id = arrs["id"]
-    variant_contig: ArrayLike = arrs["chrom"].astype(contig_dtype)
-    variant_contig, variant_contig_names = encode_contigs(variant_contig)
-    variant_contig_names = list(variant_contig_names)
-    variant_position = arrs["pos"]
-    variant_allele = da.hstack((arrs["a1"][:, np.newaxis], arrs["a2"][:, np.newaxis]))
+        variant_id = arrs["id"]
+        variant_contig: ArrayLike = arrs["chrom"].astype(contig_dtype)
+        variant_contig, variant_contig_names = encode_contigs(variant_contig)
+        variant_contig_names = list(variant_contig_names)
+        variant_position = arrs["pos"]
+        variant_allele = da.hstack(
+            (arrs["a1"][:, np.newaxis], arrs["a2"][:, np.newaxis])
+        )
 
-    call_genotype_probability = da.from_array(
-        bgen_reader,
-        chunks=chunks,
-        lock=lock,
-        fancy=False,
-        asarray=False,
-        name=f"{bgen_reader.name}:read_bgen:{path}",
-    )
-    call_dosage = _to_dosage(call_genotype_probability)
+        call_genotype_probability = da.from_array(
+            bgen_reader,
+            chunks=chunks,
+            lock=lock,
+            fancy=False,
+            asarray=False,
+            name=f"{bgen_reader.name}:read_bgen:{path}",
+        )
+        call_dosage = _to_dosage(call_genotype_probability)
 
-    ds: Dataset = create_genotype_dosage_dataset(
-        variant_contig_names=variant_contig_names,
-        variant_contig=variant_contig,
-        variant_position=variant_position,
-        variant_allele=variant_allele,
-        sample_id=sample_id,
-        call_dosage=call_dosage,
-        call_genotype_probability=call_genotype_probability,
-        variant_id=variant_id,
-    )
+        ds: Dataset = create_genotype_dosage_dataset(
+            variant_contig_names=variant_contig_names,
+            variant_contig=variant_contig,
+            variant_position=variant_position,
+            variant_allele=variant_allele,
+            sample_id=sample_id,
+            call_dosage=call_dosage,
+            call_genotype_probability=call_genotype_probability,
+            variant_id=variant_id,
+        )
 
-    return ds
+        return ds
 
 
 def _default_sample_ids(path: PathType) -> ArrayLike:

--- a/sgkit/io/plink/plink_reader.py
+++ b/sgkit/io/plink/plink_reader.py
@@ -2,6 +2,7 @@
 from pathlib import Path
 from typing import Any, Dict, List, MutableMapping, Optional, Tuple, Union
 
+import dask
 import dask.array as da
 import dask.dataframe as dd
 import numpy as np
@@ -245,64 +246,65 @@ def read_plink(
             f"{path}.{ext}" for ext in ["bed", "bim", "fam"]
         ]
 
-    # Load axis data first to determine dimension sizes
-    df_fam = read_fam(fam_path, sep=fam_sep)  # type: ignore[arg-type]
-    df_bim = read_bim(bim_path, sep=bim_sep)  # type: ignore[arg-type]
+    with dask.config.set({"dataframe.convert-string": False}):
+        # Load axis data first to determine dimension sizes
+        df_fam = read_fam(fam_path, sep=fam_sep)  # type: ignore[arg-type]
+        df_bim = read_bim(bim_path, sep=bim_sep)  # type: ignore[arg-type]
 
-    if persist:
-        df_fam = df_fam.persist()
-        df_bim = df_bim.persist()
+        if persist:
+            df_fam = df_fam.persist()
+            df_bim = df_bim.persist()
 
-    arr_fam = dataframe_to_dict(df_fam, dtype=FAM_ARRAY_DTYPE)
-    arr_bim = dataframe_to_dict(df_bim, dtype=BIM_ARRAY_DTYPE)
+        arr_fam = dataframe_to_dict(df_fam, dtype=FAM_ARRAY_DTYPE)
+        arr_bim = dataframe_to_dict(df_bim, dtype=BIM_ARRAY_DTYPE)
 
-    # Load genotyping data
-    call_genotype = da.from_array(
-        # Make sure to use asarray=False in order for masked arrays to propagate
-        BedReader(bed_path, (len(df_bim), len(df_fam)), count_A1=count_a1),  # type: ignore[arg-type]
-        chunks=chunks,
-        # Lock must be true with multiprocessing dask scheduler
-        # to not get bed-reader errors (it works w/ threading backend though)
-        lock=lock,
-        asarray=False,
-        name=f"bed_reader:read_plink:{bed_path}",
-    )
+        # Load genotyping data
+        call_genotype = da.from_array(
+            # Make sure to use asarray=False in order for masked arrays to propagate
+            BedReader(bed_path, (len(df_bim), len(df_fam)), count_A1=count_a1),  # type: ignore[arg-type]
+            chunks=chunks,
+            # Lock must be true with multiprocessing dask scheduler
+            # to not get bed-reader errors (it works w/ threading backend though)
+            lock=lock,
+            asarray=False,
+            name=f"bed_reader:read_plink:{bed_path}",
+        )
 
-    # If contigs are already integers, use them as-is
-    if bim_int_contig:
-        variant_contig = arr_bim["contig"].astype("int16")
-        variant_contig_names = da.unique(variant_contig).astype(str)
-        variant_contig_names = list(variant_contig_names.compute())
-    # Otherwise create index for contig names based
-    # on order of appearance in underlying .bim file
-    else:
-        variant_contig, variant_contig_names = encode_array(arr_bim["contig"].compute())  # type: ignore
-        variant_contig = variant_contig.astype("int16")
-        variant_contig_names = list(variant_contig_names)
+        # If contigs are already integers, use them as-is
+        if bim_int_contig:
+            variant_contig = arr_bim["contig"].astype("int16")
+            variant_contig_names = da.unique(variant_contig).astype(str)
+            variant_contig_names = list(variant_contig_names.compute())
+        # Otherwise create index for contig names based
+        # on order of appearance in underlying .bim file
+        else:
+            variant_contig, variant_contig_names = encode_array(arr_bim["contig"].compute())  # type: ignore
+            variant_contig = variant_contig.astype("int16")
+            variant_contig_names = list(variant_contig_names)
 
-    variant_position = arr_bim["pos"]
-    a1: ArrayLike = arr_bim["a1"].astype("str")
-    a2: ArrayLike = arr_bim["a2"].astype("str")
+        variant_position = arr_bim["pos"]
+        a1: ArrayLike = arr_bim["a1"].astype("str")
+        a2: ArrayLike = arr_bim["a2"].astype("str")
 
-    # Note: column_stack not implemented in Dask, must use [v|h]stack
-    variant_allele = da.hstack((a1[:, np.newaxis], a2[:, np.newaxis]))
-    variant_allele = variant_allele.astype("S")
-    variant_id = arr_bim["variant_id"]
+        # Note: column_stack not implemented in Dask, must use [v|h]stack
+        variant_allele = da.hstack((a1[:, np.newaxis], a2[:, np.newaxis]))
+        variant_allele = variant_allele.astype("S")
+        variant_id = arr_bim["variant_id"]
 
-    sample_id = arr_fam["member_id"]
+        sample_id = arr_fam["member_id"]
 
-    ds = create_genotype_call_dataset(
-        variant_contig_names=variant_contig_names,
-        variant_contig=variant_contig,
-        variant_position=variant_position,
-        variant_allele=variant_allele,
-        sample_id=sample_id,
-        call_genotype=call_genotype,
-        variant_id=variant_id,
-    )
+        ds = create_genotype_call_dataset(
+            variant_contig_names=variant_contig_names,
+            variant_contig=variant_contig,
+            variant_position=variant_position,
+            variant_allele=variant_allele,
+            sample_id=sample_id,
+            call_genotype=call_genotype,
+            variant_id=variant_id,
+        )
 
-    # Assign PLINK-specific pedigree fields
-    return ds.assign({f"sample_{f}": (DIM_SAMPLE, arr_fam[f]) for f in arr_fam})
+        # Assign PLINK-specific pedigree fields
+        return ds.assign({f"sample_{f}": (DIM_SAMPLE, arr_fam[f]) for f in arr_fam})
 
 
 def plink_to_zarr(

